### ISSUE TITLE
Failing tests on PG15.

### ISSUE THREAD

**Issue by sb230132** [timescale/timescaledb#4835]
### What type of bug is this?

Incorrect result

### What subsystems and features are affected?

Distributed hypertable

### What happened?

TimescaleDB when compiled against PG15 causes many tests to produce different results based on PG version.
ex:
```
+++ /home/runner/work/timescaledb/timescaledb/build/tsl/test/shared/results/dist_parallel_agg.out	2022-10-14 13:19:24.114062850 +0000
@@ -35,74 +35,28 @@
 :analyze
 select count(*) from metrics_dist1;
 QUERY PLAN
- Custom Scan (DataNodeScan) (actual rows=1 loops=1)
-   Output: (count(*))
-   Relations: Aggregate on (public.metrics_dist1)
-   Data node: data_node_1
-   Fetcher Type: Row by row
-   Chunks: _dist_hyper_X_X_chunk, _dist_hyper_X_X_chunk
-   Remote SQL: SELECT count(*) FROM public.metrics_dist1 WHERE _timescaledb_internal.chunks_in(public.metrics_dist1.*, ARRAY[..])
-   Remote EXPLAIN:
-     Finalize Aggregate (actual rows=1 loops=1)
-       Output: count(*)
-       ->  Gather (actual rows=2 loops=1)
-             Output: (PARTIAL count(*))
-             Workers Planned: 1
-             Workers Launched: 1
-             ->  Partial Aggregate (actual rows=1 loops=2)
-                   Output: PARTIAL count(*)
-                   Worker 0: actual rows=1 loops=1
-                   ->  Parallel Append (actual rows=10000 loops=2)
-                         Worker 0: actual rows=0 loops=1
-                         ->  Parallel Seq Scan on _timescaledb_internal._dist_hyper_X_X_chunk (actual rows=17990 loops=1)
-                         ->  Parallel Seq Scan on _timescaledb_internal._dist_hyper_X_X_chunk (actual rows=2010 loops=1)
-
-(22 rows)
+ Aggregate (actual rows=1 loops=1)
+   Output: count(*)
+   ->  Result (actual rows=0 loops=1)
+         One-Time Filter: false
+(4 rows)
```

### TimescaleDB version affected

2.9.0

### PostgreSQL version used

15.0

### What operating system did you use?

Mac OS X

### What installation method did you use?

Source

### What platform did you run on?

On prem/Self-hosted

### How can we reproduce the bug?

```bash
Please run below regresscheck tests.

make installcheck TESTS="parallel plan_hashagg query partialize_finalize  dist_distinct dist_distinct_pushdown dist_fetcher_type dist_gapfill dist_parallel_agg dist_queries dist_remote_error jit-15 debug_notice dist_chunk"
```

**Comment by sb230132** [comment]
Fix from PR#4841 has fixed some of the tests. However there are still few tests which are failing like below:
parallel
plan_hashagg
query
partialize_finalize
dist_distinct
dist_fetcher_type
dist_remote_error
jit-15

**Comment by sb230132** [comment]
Issue with plan_hashagg is that error message has been changed in PG15 because of which the test fails.
```
diff -u /Volumes/Work/timescaledb/test/expected/plan_hashagg.out /Volumes/Work/timescaledb/build15/test/results/plan_hashagg.out
--- /Volumes/Work/timescaledb/test/expected/plan_hashagg.out	2022-11-07 21:36:00
+++ /Volumes/Work/timescaledb/build15/test/results/plan_hashagg.out	2022-11-08 12:02:28
@@ -326,5 +326,5 @@
 (7 rows)
 
 \set ECHO none
-psql:include/plan_hashagg_query.sql:60: ERROR:  timestamp units "invalid" not recognized
-psql:include/plan_hashagg_query.sql:60: ERROR:  timestamp units "invalid" not recognized
+psql:include/plan_hashagg_query.sql:60: ERROR:  unit "invalid" not recognized for type timestamp without time zone
+psql:include/plan_hashagg_query.sql:60: ERROR:  unit "invalid" not recognized for type timestamp without time zone
```

**Comment by sb230132** [comment]
Fixed dist_remote_error jit-15 in #4990

Yet to fix are:
plan_hashagg
dist_fetcher_type
dist_partial_agg

**Comment by sb230132** [comment]
There are separate issues to track above failures. Thus closing this issue.
dist_move_chunk - Refer #4972
remote_txn - Refer #5037
